s3/tests/integration/test-charm-s3/src/s3_lib.py

```diff
@@ -227,7 +229,11 @@ def __init__(self, charm: CharmBase, relation_data: S3RequirerData):
     def _on_relation_joined_event(self, event: RelationJoinedEvent) -> None:
         """Event emitted when the S3 relation is joined."""
         logger.debug(f"S3 relation ({event.relation.name}) joined...")
-        event_data = {"bucket": self.relation_data.bucket, "path": self.relation_data.path}
+        event_data = {
+            "bucket": self.relation_data.bucket,
+            "path": self.relation_data.path,
+            "version": str(LIBAPI),
```

Comment:
This field is not necesssary here, however I've added it anyway. This might come very useful for us in the future, should we need to decide what version of LIBAPI the requirer is using. @deusebio  @welpaolo  @Gu1nness what do you think about this?

Reply:
+1 I believe it is good to include the version field, it should make handling compatibility between requirer and provider a bit easier if both sides communicate the version, and we would ever have compatibility edge cases to care about